src/query/service/src/interpreters/interpreter_table_modify_column.rs

```diff
@@ -94,21 +108,143 @@ impl ModifyTableColumnInterpreter {
         };
         column_mask_policy.insert(self.plan.column.clone(), mask_name);
         new_table_meta.column_mask_policy = Some(column_mask_policy);
-        Ok(new_table_meta)
+
+        let table_id = table_info.ident.table_id;
+        let table_version = table_info.ident.seq;
+
+        let req = UpdateTableMetaReq {
+            table_id,
+            seq: MatchSeq::Exact(table_version),
+            new_table_meta,
+            copied_files: None,
+            deduplicated_label: None,
+        };
+
+        let res = catalog.update_table_meta(table_info, req).await?;
+
+        if let Some(share_table_info) = res.share_table_info {
+            save_share_table_info(
+                &self.ctx.get_tenant(),
+                self.ctx.get_data_operator()?.operator(),
+                share_table_info,
+            )
+            .await?;
+        }
+        Ok(PipelineBuildResult::create())
+    }
+
+    // Set data column type.
+    async fn do_set_data_type(
+        &self,
+        table: &Arc<dyn Table>,
+        type_name: &TypeName,
+    ) -> Result<PipelineBuildResult> {
+        let schema = table.schema().as_ref().clone();
+        let table_info = table.get_table_info();
+        if let Ok(i) = schema.index_of(&self.plan.column) {
+            // Check if this column is referenced by computed columns.
+            let field: DataField = schema.field(i).into();
+            if field.computed_expr().is_none() {
+                check_referenced_computed_columns(
```

Comment:
The `check_referenced_computed_columns` function needs to be rewritten to handle the case where the column data type changes